Quadriga/src/main/java/edu/asu/spring/quadriga/service/textfile/impl/TextFileManager.java

```diff
@@ -0,0 +1,109 @@
+
+import edu.asu.spring.quadriga.dao.textfile.ITextFileDAO;
+import edu.asu.spring.quadriga.dao.workbench.IProjectWorkspaceDAO;
+import edu.asu.spring.quadriga.domain.workspace.ITextFile;
+import edu.asu.spring.quadriga.dto.TextFileDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.textfile.ITextFileManager;
+import edu.asu.spring.quadriga.service.textfile.mapper.ITextFileShallowMapper;
+
+@PropertySource(value = "classpath:/user.properties")
+@Service
+public class TextFileManager implements ITextFileManager {
+
+    @Autowired
+    private ITextFileDAO txtFileDAO;
+
+    @Autowired
+    private IProjectWorkspaceDAO projWSDAO;
+
+    @Autowired
+    private Environment env;
+    
+    @Autowired
+    private ITextFileShallowMapper tfSMapper;
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * edu.asu.spring.quadriga.service.textfile.ITextFileService#saveTextFile(
+     * edu.asu.spring.quadriga.domain.workspace.ITextFile) Service layer method
+     * to handle Text Management Operations
+     */
+    @Override
+    public boolean saveTextFile(ITextFile txtFile) throws QuadrigaStorageException, IOException {
+        String txtId = (txtFileDAO.generateUniqueID());
+        txtFile.setTextId(txtId);
+        TextFileDTO txtFileDTO = tfSMapper.getTextFileDTO(txtFile);
+        txtFileDTO.setTextId(txtId);
+        boolean fsStatus = saveTextFileLocal(txtFile);
+        boolean dbStatus = saveTextFileDB(txtFileDTO);
+        return fsStatus && dbStatus;
+    }
+
+    /**
+     * @param txtFile
+     *            TextFile object to be updated in the database
+     * @return
+     * returns true if file is successfully saved else returns false.
+     * @throws QuadrigaStorageException
+     */
+    private boolean saveTextFileDB(TextFileDTO tfDTO) throws QuadrigaStorageException {   
+        System.out.println(tfDTO.getTextId());
+        return txtFileDAO.saveTextFileDTO(tfDTO);
+    }
+
+    /**
+     * @param txtFile
+     *            TextFile object to be updated in the FileSystem
+     * @return
+     * @throws IOException
+     */
+    private boolean saveTextFileLocal(ITextFile txtFile) throws IOException {
+
+        String saveDir = env.getProperty("textfile.location");
+        System.out.println(env.getProperty("textfile.location"));
+        String filePath = saveDir + "/" + txtFile.getTextId();
+        File saveTxtFile;
+        File dirFile = new File(filePath);
+        if (!dirFile.exists()) {
+            dirFile.mkdir();
+        }
+        String fileName = txtFile.getFileName();
+        if (fileName.contains(".")) {
+            saveTxtFile = new File(filePath + "/" + fileName);
+        } else {
+            saveTxtFile = new File(filePath + "/" + fileName + ".txt");
+        }
+        FileWriter fw = new FileWriter(saveTxtFile);
+        fw.write(txtFile.getFileContent());
+        File propFile = new File(filePath + "/meta.properties");
```

Comment:
creating the metadata file should be in its own method and then just called here.